packages/flutter_tools/lib/src/runner/flutter_command.dart

```diff
@@ -1907,7 +1907,10 @@ Run 'flutter -h' (or 'flutter <command> -h') for available flutter commands and
         // The preview device does not currently support any plugins.
         allowedPlugins = PreviewDevice.supportedPubPlugins;
       }
-      await project.regeneratePlatformSpecificTooling(allowedPlugins: allowedPlugins);
+      await project.regeneratePlatformSpecificTooling(
+        allowedPlugins: allowedPlugins,
+        releaseMode: getBuildMode().isRelease,
```

Comment:
We want to provide this only when the feature flag`--explicit-package-dependencies` is enabled, otherwise we'll start stripping plugin registrations.